### PR TITLE
Error handling for handlers

### PR DESCRIPTION
**Pull request by joestubbs**
The main change in this PR is to update the request handlers to all use TagentError, and to move TagentError to representations. 

Additionally, this PR fixes the upload and download handlers that were not converted to the new actix-web v4 application state object, and it adds some logging and code clean up.

Finally, I added a Makefile to help me remember the commands I need to run. I don't know if Makefiles are "the Rust way" so if not we can do something else.

Let me know your thoughts on any of this!

### CODE REVIEW

**Comment by waltermoreira** [comment]
Hope you don't mind, I pushed a commit to fix that import that was giving us problem. I'll keep taking a look...

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+	cargo build
+
+test:
+	cargo test -- --test-threads=1
+
+run-local:
+	cargo run
+
+lint:
+	cargo clippy
```
Comment (waltermoreira) [inline Makefile]: I feel that Makefiles, particularly with such a clean structure as this one, are great. I've seen several other task runners, but I'm not sure there is another standard one. I've seen `cargo make` (https://github.com/sagiegurari/cargo-make) but I'm not sure it is worth to explore at this time. Maybe if this Makefile starts getting too big?

The only thing I'd suggest here is:
```suggestion
	cargo clippy  --all-features --tests
```
so it also checks the tests.

File: tagent/src/handlers.rs [hunk 2]
```diff
@@ -105,7 +100,7 @@ pub fn get_local_listing(full_path: PathBuf) -> Vec<String> {
     result
 }
 
-type FileListHttpRsp = Either<HttpResponse, web::Json<FileListingRsp>>;
+type FileListHttpRsp = Result<web::Json<FileListingRsp>, TagentError>;
```
Comment (waltermoreira) [inline tagent/src/handlers.rs:101]: Love it! Not a big fan of those `Either` that were there before, to be honest. But the explicit error makes it so much better now.

File: tagent/src/representations.rs [hunk 3]
```diff
@@ -38,3 +40,35 @@ pub struct FileUploadRsp {
...
+            message: m.to_string(),
+            version: v.to_string(),
+            result: String::from("none"),
+        };
+        let body = serde_json::to_value(&r).unwrap().to_string();
+        HttpResponse::BadRequest().body(body)
+    }
+}
+
+pub fn make_tagent_error(message: String, version: String) -> TagentError {
```
Comment (waltermoreira) [inline tagent/src/representations.rs]: This `make_tagent_error` was my own naming, and I feel it's not up to par. I think the rust way is to make it a `TagentError::new` constructor. Check the branch `dev-wm` and compare it with this one (`git diff dev dev-wm`). Would you be up to change it? If you like it, we can add it to the PR (`git checkout dev; git merge dev-wm`).

Comment (joestubbs) [reply tagent/src/representations.rs]: totally agree!

**Pull request merged**